Automattic/jetpack
# Avoid local eslint errors from e2e dirs

#### Changes proposed in this Pull Request:

We have the e2e test directories in separate pnpm workspaces because
they require large downloads for playwright that are not normally needed
in local development. But that can cause problems when trying to run
eslint as the plugins referenced from eslintrc are not available.

To avoid that, let's adjust the eslint configurations in the e2e dirs to
detect this situation and avoid trying to lint at all.

#### Jetpack product discussion


p1643055734132300/1643054245.126200-slack-CBG1CP4EN
p1642686982001900-slack-CBG1CP4EN
p1636751680290600-slack-CBG1CP4EN
p1634153125426300-slack-CBG1CP4EN
p1643631857393939/1634153125.426300-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Clone a fresh checkout. `pnpm install && pnpm lint-required` should succeed, with a note about eslint being disabled in `projects/plugins/boost/tests/e2e`.
* Do `CI=true pnpm lint-required`. This time it should fail with an error about being unable to find `jetpack-e2e-commons/.eslintrc.cjs`.
* Do `pnpm --dir projects/plugins/boost/tests/e2e install`, then run `pnpm lint-required` again. It should succeed, this time with no note about eslint being disabled.
